### DISCUSSION TITLE
1.4 Sidebar Sorting

### DISCUSSION THREAD

**Discussion by james-d-elliott** [thuliteio/doks#1210] [Q&A]
Hey all, love the new auto-menu it works great mostly. One thing I'm looking for is figuring out how to group by weight then sort by title within each group. Has anyone accomplished this before? Currently it appears to be by weight then by date.

My experiments so far have been with [section-menu.html](https://github.com/gethyas/doks-core/blob/main/layouts/partials/sidebar/section-menu.html) and [render-section-menu.html](https://github.com/gethyas/doks-core/blob/main/layouts/partials/sidebar/render-section-menu.html) but can't quite get it right.

**Comment by h-enk** [comment]
Thanks! Just change the default sort order — https://github.com/gethyas/doks-core/blob/main/layouts/partials/sidebar/render-section-menu.html#L92 — to:

```html
        {{- range .ByTitle }}
          {{- template "walk" (dict "node" . "currentPage" $currentPage) }}
        {{- end }}
```

That should work for what you want — see also [Sort content](https://gohugo.io/templates/lists/#sort-content)

![Snag_14d0e2c3](https://github.com/gethyas/doks/assets/3902872/fc90d936-75f8-47cb-8d17-262c146bf327)

**Comment by james-d-elliott** [comment]
Aha! Not exactly what I was after (probably didn't explain it well) but managed to get it, the pointer was instrumental in finding where to do it exactly.

Example:

```
        {{- range .GroupBy "Weight" "asc" }}
        {{- range .ByTitle }}
          {{- template "walk" (dict "node" . "currentPage" $currentPage) }}
        {{- end }}
        {{- end }}
```

**Reply by h-enk** [reply]
Ah, super — glad you got it!

**Comment by sarviun** [comment]
Not messing with brand new topic so I ask here. I want to edit [render-section-menu.html](https://github.com/gethyas/doks-core/blob/main/layouts/partials/sidebar/render-section-menu.html) as well cause I have done some editing in   `layouts/partials/sidebar/auto-default-menu.html` in previous version. Since it is a node package module in node_modules now, how can I avoid overwriting the changes when deploying or whenever `npm updates` runs? Sorry for stupid question, not a frontend expert. I googled some answers but I don't like any of them.

**Reply by james-d-elliott** [reply]
You can place it directly in the layouts/partials/sidebar/ directory which you may need to create rather than the node modules.

**Reply by sarviun** [reply]
Really that simple! Thank you!
